src/__test__/App.test.js

```diff
@@ -0,0 +1,63 @@
+import { render, screen } from '@testing-library/react';
+import { MemoryRouter, BrowserRouter } from 'react-router-dom';
+import { Provider } from 'react-redux';
+import store from '../redux/configurStor';
+
+import App from '../App';
+import HomePage from '../Components/HomePage';
+import DetailsPage from '../Components/DetailsPage';
+import HomepageHeader from '../Components/HomePageHeader';
+import DetailsPageHeader from '../Components/DetailsPageHeader';
+
+jest.mock('../Components/HomePage');
+jest.mock('../Components/DetailsPage');
+jest.mock('../Components/DetailsPageHeader');
+
+describe('Test app.js with pages', () => {
+  test('should render the Home page header', () => {
+    render(
+      <BrowserRouter>
+        <HomepageHeader />
+      </BrowserRouter>,
+    );
+    const txt = screen.getByText('Countries Of The World', { exact: false });
+    expect(txt).toBeInTheDocument();
+  });
+
+  test('should render the Details page header', () => {
+    DetailsPageHeader.mockImplementation(() => <h1>Hello From Details Page Header</h1>);
+    render(
+      <BrowserRouter>
+        <DetailsPageHeader />
+      </BrowserRouter>,
+    );
+    const txt = screen.getByText('Hello From Details Page Header');
+    expect(txt).toBeInTheDocument();
+  });
+
+  test('should render the homepage', () => {
+    HomePage.mockImplementation(() => <h1>Hello From Home Page</h1>);
+    render(
+      <MemoryRouter initialEntries={['/']}>
+        <Provider store={store}>
+          <App />
+        </Provider>
+      </MemoryRouter>,
+    );
+    const txt = screen.getByText('Hello From Home Page');
+    expect(txt).toBeInTheDocument();
+  });
+
+  test('should render the Details Page', () => {
+    DetailsPage.mockImplementation(() => <h1>Hello From Details Page</h1>);
+    render(
+      <MemoryRouter initialEntries={['/details/Bangladesh']}>
+        <Provider store={store}>
+          <App />
+        </Provider>
```

Comment:
- [ ] Nice adding the integration tests but you should also add unit tests for pure redux functions as required in order to complete the requirements 
- Screenshot of the requirements
<img width="640" alt="Screenshot 2023-02-14 at 15 47 19" src="https://user-images.githubusercontent.com/61790587/218771992-00e8a203-3d75-4179-b95f-fbc127a0b0f6.png">

Reply:
I have successfully added the unit tests for the redux function. Thanks for the observation.